rules/no-typeof-undefined.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+const {findVariable} = require('eslint-utils');
+const {matches} = require('./selectors/index.js');
+
+const MESSAGE_ID = 'NO_TYPEOF_UNDEFINED';
+const messages = {
+	[MESSAGE_ID]: 'Prefer `{{operator}}` operator instead of typeof with \'undefined\'.',
+};
+
+const selector = [
+	'BinaryExpression',
+	matches(['[right.value=undefined]', '[left.value=undefined]']),
+	' UnaryExpression[operator=typeof]',
```

Comment:
The `UnaryExpression` should  be  a direct child of `BinaryExpression`.

```suggestion
	' > UnaryExpression[operator="typeof"]',
```